### ISSUE TITLE
Trading - e2e tests

### ISSUE THREAD

**Issue by keraf** [trezor/trezor-suite#3689]
Cover trading section by E2E tests

> Current state of automated test coverage in suite-web/e2e: There are 2 tests currently enabled, which are covering the invity powered trading page in `packages/suite-web/e2e/tests/coinmarket/buy.test.ts`
> 
> * Should buy crypto best offer successfully - basic buy flow (✅  Passing)
> * Should buy crypto with comparing all offers successfully - flow with comparing different offers (✅  Passing)
> 
> Both tests are using mocked responses for the invity backend (I suppose that it's fine, but not really E2E). As most tests in the suite, these also don't use common patterns like page objects for better abstraction.
> 
> I couldn't find any other currently enabled tests for trading flows e.g. sell, swap, dca) or for other coins than BTC.
> 
> Additionally there is a more complex scenario for exchanging coins in `packages/suite-web/e2e/tests/coinmarket/exchange.test.ts` but it's currently skipped. I tried running it and it failed ❌ so it needs fixing. And lastly in `packages/suite-web/e2e/tests/dashboard/dashboard.test.ts` there is just very basic checking that the trading form actually has all the input fields.

**Comment by martinboehm** [comment]
Totally agree. And finally we also have resources to create them @LukasRada.

**Comment by tsusanka** [comment]
@LukasRada @martinboehm how is this looking?

**Comment by tomasklim** [comment]
Would be nice to have e2e tests on Invity forms to check that #4650 does not break anything

**Comment by tomasklim** [comment]
Any progress @LukasRada @martinboehm?

**Comment by LukasRada** [comment]
We're actively looking for tester so he/she can finish the e2e tests. When the tester is in our team, we allocate him/her on it. Our main priority for now is to finish DCA.

**Comment by HajekOndrej** [comment]
Current state of automated test coverage in suite-web/e2e:
There are 2 tests currently enabled, which are covering the invity powered trading page in `packages/suite-web/e2e/tests/coinmarket/buy.test.ts`

- Should buy crypto best offer successfully - basic buy flow (✅  Passing)
- Should buy crypto with comparing all offers successfully - flow with comparing different offers (✅  Passing)

Both tests are using mocked responses for the invity backend (I suppose that it's fine, but not really E2E). As most tests in the suite, these also don't use common patterns like page objects for better abstraction. 

I couldn't find any other currently enabled tests for trading flows e.g. sell, swap, dca) or for other coins than BTC.


Additionally there is a more complex scenario for exchanging coins in `packages/suite-web/e2e/tests/coinmarket/exchange.test.ts` but it's currently skipped. I tried running it and it failed ❌ so it needs fixing. And lastly in `packages/suite-web/e2e/tests/dashboard/dashboard.test.ts` there is just very basic checking that the trading form actually has all the input fields.

**Comment by Vere-Grey** [comment]
We have covered now several happy paths flow for both BUY, SELL and SWAP
To be review with Klima, once that is done we will discuss if want to keep this opened or not. I have already followup Issue for advance scenarios.

**Comment by HajekOndrej** [comment]
Main trading scenarios are already covered. Closing.
